src/ui/components/SpecialDistribution.js

```diff
@@ -1,38 +1,79 @@
 // @flow
-
 import React, {useState} from "react";
-import {div, fromInteger, fromFloatString, ZERO} from "../../ledger/grain";
-import {type IdentityId} from "../../ledger/identity";
+import {Button, Container, TextField} from "@material-ui/core";
+import {Alert} from "@material-ui/lab";
+import {makeStyles} from "@material-ui/core/styles";
+import {div, fromFloatString, lt, ZERO} from "../../ledger/grain";
 import {type Account} from "../../ledger/ledger";
 import AccountDropdown from "./AccountSelector";
 import {computeAllocation} from "../../ledger/grainAllocation";
 import * as uuid from "../../util/uuid";
 import type {TimestampMs} from "../../util/timestamp";
 import {useLedger} from "../utils/LedgerContext";
 
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "80%",
+    minWidth: "700px",
+    margin: "0 auto",
+    padding: "0 5em 5em",
+  },
+  arrowBody: {
+    color: theme.palette.text.primary,
+    flex: 1,
+    background: theme.palette.background.paper,
+    padding: "5px 20px",
+    display: "flex",
+    alignItems: "center",
+  },
+  triangle: {
+    width: 0,
+    height: 0,
+    background: theme.palette.background,
+    borderTop: "30px solid transparent",
+    borderBottom: "30px solid transparent",
+    borderLeft: `30px solid ${theme.palette.background.paper}`,
+  },
+  dropdownWrapper: {
+    border: `1px solid ${theme.palette.text.primary}`,
+    borderRadius: "50px",
+  },
+  centerRow: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+  },
+  element: {flex: 1, margin: "20px"},
+  arrowInput: {display: "inline-block"},
+  headerText: {color: theme.palette.text.primary},
+}));
+
 export const SpecialDistribution = () => {
   const {ledger, updateLedger} = useLedger();
 
+  const classes = useStyles();
   const [credTimestamp, setCredTimestamp] = useState<TimestampMs>(+Date.now());
-  const [recipient, setRecipient] = useState<IdentityId | null>(null);
-  const [amount, setAmount] = useState<string>("0");
+  const [recipient, setRecipient] = useState<Account | null>(null);
```

Comment:
Why would we switch this to have an account as state? Note that this implies the account state in the component can desynchronize from the account in the ledger. It's nice to use the ledger as the source of truth for all account data.

Reply:
The main reason I did this is we do the same thing in the transfer grain UI, and I wanted access to the name in the alert. Should I just have a separate state variable for the recipientName? I had two variables before in the transfer grain UI and you had me just set the account in state there. Is there a reason that we should treat these two UIs differently in that regard?